plugins/services/src/js/utils/GraphQLTaskUtil.js

```diff
@@ -96,15 +99,42 @@ function mergeHostname(task) {
   return task;
 }
 
+function mergeData(task, flattenTaskTree) {
+  // Merge version from Marathon
+  task = mergeVersion(task, flattenTaskTree);
+  // Get Health from Mesos first, and fallback on Marathon
+  task = mergeHealth(task, flattenTaskTree);
+  // Merge hostname if we can find it
+  task = mergeHostname(task);
+
+  return task;
+}
+
+function buildFlatServiceTree(serviceTree) {
+  return serviceTree.reduceItems((memo, node) => {
+    if (node instanceof ServiceTree) {
+      return memo;
+    }
+
+    const taskData = {
+      version: node.version,
+      health: node.healthCheckResults
+    };
+
+    // console.log("cache ", node.id);
+    memo[node.id] = taskData;
```

Comment:
☝️ please use id getter instead of directly accessing the value.

Reply:
node.getId() you mean?